src/js/components/form-elements/TableBody.jsx

```diff
@@ -1,17 +1,41 @@
 import React from 'react';
 
 import PropTypes from 'prop-types';
+import { getTranslate } from 'react-localize-redux';
+import { connect } from 'react-redux';
 
 import TableRow from 'components/form-elements/TableRow';
-
+import Spinner from 'components/spinner/Spinner';
+import { translateWithDefaultMessage } from 'utils/Translate';
 
 const TableBody = (props) => {
   const {
-    fieldsConfig, properties, fields, tableRef = () => {},
+    translate,
+    fieldsConfig,
+    properties,
+    fields,
+    tableRef = () => {
+    },
     addRow = (row = {}) => fields.push(row),
   } = props;
   const RowComponent = properties.subfield ? TableRow : fieldsConfig.rowComponent || TableRow;
 
+  const {
+    emptySubstitutions,
+    loadingSubstitutions,
+  } = properties;
```

Comment:
I think you should add a comment about usage of those properties and also about the two ifs below, as `TableBody` is a component we use in many places and this particular case will only be used for substitution modal

Reply:
I also think that such a request might come in some other places/modals, so maybe we could think about some "generic" naming here? @awalkowiak what do you think?

Reply:
@kchelstowski @alannadolny yes, this definitely has to be generic